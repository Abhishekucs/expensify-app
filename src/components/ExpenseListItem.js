import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'

const ExpenseListItem = ({ id, description, createdAt, amount }) => (
    <div>
        <Link to={`/edit/${id}`}><h1>{description}</h1></Link>
        <p>
            <span>Rs</span>{new Intl.NumberFormat('en-IN').format(amount)}
            -
            {moment(createdAt).format('MMM Do, YYYY')}
        </p>
    </div>
)

export default ExpenseListItem