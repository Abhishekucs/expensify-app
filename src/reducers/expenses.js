// Expense Reducer

const expenseReducerInitialState = []

const expenseReducer = (state = expenseReducerInitialState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ]
        case 'REMOVE_EXPENSE':
            return state.filter((expense) => {
                return expense.id !== action.id
            })
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if(expense.id === action.id) {
                    return {
                        ...expense,
                        ...action.updates
                    }
                }else {
                    return expense
                }
            })
        case 'SET_EXPENSES':
            return action.expenses
        default:
            return state
    }
}

export default expenseReducer