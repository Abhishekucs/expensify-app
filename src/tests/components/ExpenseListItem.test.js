import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import expenses from '../fixtures/expenses'
import toJSON from 'enzyme-to-json'

test('should render Expense List Item', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>)
    expect(toJSON(wrapper)).toMatchSnapshot()
})