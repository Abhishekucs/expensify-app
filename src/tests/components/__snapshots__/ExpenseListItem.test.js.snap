// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Expense List Item 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h1>
      Rent
    </h1>
  </Link>
  <p>
    <span>
      Rs
    </span>
    3,500
    -
    Jan 1st, 1970
  </p>
</div>
`;
